poller-lambdas/localRun.ts

```diff
@@ -0,0 +1,131 @@
+	ingestionQueue: SendMessageCommand[],
+) => {
+	console.log('---------------------------------------------');
+	console.log(`POLLER INVOCATION STARTED ${new Date().toISOString()}`);
+	console.log('---------------------------------------------');
+
+	await handler({
+		Records: [
+			{
+				body: input,
+			},
+		],
+	});
+
+	const nextQueueItem = ownQueue.shift()!.input;
+	const inputForNext = nextQueueItem.MessageBody!;
+	const maybeDelaySeconds = nextQueueItem.DelaySeconds;
+
+	console.log('---------------------------------------------');
+	console.log(`POLLER INVOCATION COMPLETED ${new Date().toISOString()}`);
+	if (maybeDelaySeconds) {
+		console.log(`  DelaySeconds: ${maybeDelaySeconds}`);
+	}
+	console.log(`  input for next invocation: ${inputForNext}`);
+	while (ingestionQueue.length > 0) {
+		const nextIngestionItem = ingestionQueue.shift()!.input;
+		console.log(`  ITEM SENT TO INGESTION QUEUE:`, nextIngestionItem);
+	}
+	console.log('---------------------------------------------');
+
+	const { nextAction } = (await prompts({
+		type: 'select',
+		name: 'nextAction',
+		message: 'What next?',
+		choices: [
+			...(maybeDelaySeconds
+				? [
+						{
+							title: `invoke after ${maybeDelaySeconds}s`,
+							value() {
+								console.log(
+									`Waiting ${maybeDelaySeconds}s before invoking again...`,
+								);
+								setTimeout(
+									() =>
+										void fakeInvoke(
+											handler,
+											inputForNext,
+											ownQueue,
+											ingestionQueue,
+										),
+									maybeDelaySeconds * 1000,
+								);
+							},
+						},
+					]
+				: []),
+			{
+				title: 'Invoke immediately',
+				value() {
+					void fakeInvoke(handler, inputForNext, ownQueue, ingestionQueue);
+				},
+			},
+			{ title: 'Exit', value() {} },
+		],
+	})) as { nextAction: () => void };
+
+	nextAction();
+};
+
+void (async () => {
+	if (sqs instanceof SQSClient) {
+		throw Error(
+			'SQS appears to be using a real client - this file should be using a FAKE local SQS client',
+		);
+	}
+	const ownQueueUrl = 'FAKE_OWN_QUEUE_URL';
+	process.env[POLLER_LAMBDA_ENV_VAR_KEYS.OWN_QUEUE_URL] = ownQueueUrl;
+
+	const ingestionQueueUrl = 'FAKE_INGESTION_QUEUE_URL';
+	process.env[POLLER_LAMBDA_ENV_VAR_KEYS.INGESTION_LAMBDA_QUEUE_URL] =
+		ingestionQueueUrl;
+
+	const { pollerId } = (await prompts({
+		type: 'select',
+		name: 'pollerId',
+		message: 'Which poller?',
+		choices: Object.keys(POLLERS_CONFIG).map((pollerId) => ({
+			title: pollerId,
+			value: pollerId,
+		})),
+	})) as { pollerId: PollerId };
+
+	process.env[POLLER_LAMBDA_ENV_VAR_KEYS.SECRET_NAME] = getPollerSecretName(
+		'CODE',
+		pollerId,
+	);
+
+	const handler = handlers[pollerId];
+
+	const { input } = (await prompts({
+		type: 'text',
+		name: 'input',
+		message:
+			"Do want to provide a starting input? (e.g. if this is poller is 'long polling')",
+	})) as { input: string };
+
+	await fakeInvoke(
+		handler,
+		input,
+		sqs.queueData[ownQueueUrl]!,
```

Comment:
when this line is run, afaict, there are no keys in `queueData`, so `sqs.queueData[ownQueueUrl]!,` evaluates to `undefined` and stays that way..

could potentially either pre-populate the queue with an empty array, or else do the `queueData` lookup at the time it needs to be used in the body of `fakeInvoke`?

Reply:
many thanks Pete, I've addressed this amongst other things in https://github.com/guardian/newswires/pull/83/commits/841e6f0fc31dcf920a7e416a9e33e5219464afe0